apis/kubekey/v1alpha2/kubernetes_types.go

```diff
@@ -29,21 +29,23 @@ type Kubernetes struct {
 	ApiserverCertExtraSans []string `yaml:"apiserverCertExtraSans" json:"apiserverCertExtraSans,omitempty"`
 	ProxyMode              string   `yaml:"proxyMode" json:"proxyMode,omitempty"`
 	// +optional
-	Nodelocaldns             *bool                `yaml:"nodelocaldns" json:"nodelocaldns,omitempty"`
-	EtcdBackupDir            string               `yaml:"etcdBackupDir" json:"etcdBackupDir,omitempty"`
-	EtcdBackupPeriod         int                  `yaml:"etcdBackupPeriod" json:"etcdBackupPeriod,omitempty"`
-	KeepBackupNumber         int                  `yaml:"keepBackupNumber" json:"keepBackupNumber,omitempty"`
-	EtcdBackupScriptDir      string               `yaml:"etcdBackupScript" json:"etcdBackupScript,omitempty"`
-	ContainerManager         string               `yaml:"containerManager" json:"containerManager,omitempty"`
-	ContainerRuntimeEndpoint string               `yaml:"containerRuntimeEndpoint" json:"containerRuntimeEndpoint,omitempty"`
-	ApiServerArgs            []string             `yaml:"apiserverArgs" json:"apiserverArgs,omitempty"`
-	ControllerManagerArgs    []string             `yaml:"controllerManagerArgs" json:"controllerManagerArgs,omitempty"`
-	SchedulerArgs            []string             `yaml:"schedulerArgs" json:"schedulerArgs,omitempty"`
-	KubeletArgs              []string             `yaml:"kubeletArgs" json:"kubeletArgs,omitempty"`
-	KubeProxyArgs            []string             `yaml:"kubeProxyArgs" json:"kubeProxyArgs,omitempty"`
-	FeatureGates             map[string]bool      `yaml:"featureGates" json:"featureGates,omitempty"`
-	KubeletConfiguration     runtime.RawExtension `yaml:"kubeletConfiguration" json:"kubeletConfiguration,omitempty"`
-	KubeProxyConfiguration   runtime.RawExtension `yaml:"kubeProxyConfiguration" json:"kubeProxyConfiguration,omitempty"`
+	Nodelocaldns               *bool                `yaml:"nodelocaldns" json:"nodelocaldns,omitempty"`
+	EtcdBackupDir              string               `yaml:"etcdBackupDir" json:"etcdBackupDir,omitempty"`
+	EtcdBackupPeriod           int                  `yaml:"etcdBackupPeriod" json:"etcdBackupPeriod,omitempty"`
+	KeepBackupNumber           int                  `yaml:"keepBackupNumber" json:"keepBackupNumber,omitempty"`
+	EtcdBackupScriptDir        string               `yaml:"etcdBackupScript" json:"etcdBackupScript,omitempty"`
+	ContainerManager           string               `yaml:"containerManager" json:"containerManager,omitempty"`
+	ContainerRuntimeEndpoint   string               `yaml:"containerRuntimeEndpoint" json:"containerRuntimeEndpoint,omitempty"`
+	EnableNodeFeatureDiscovery bool                 `yaml:"enableNodeFeatureDiscovery" json:"enableNodeFeatureDiscovery,omitempty"`
```

Comment:
How do you think to use a `struct` to configure the `kata` and `NodeFeatureDiscovery`?
Just like:
```
Kata:
  enable: true
  moreOptions: xxxx
NodeFeatureDiscovery:
  enable: false
```
BTW, if we use this structure, in the golang struct, the type of the `enable` field we can set as `*bool`. This allows you to distinguish between "true", "false" and "not set".

Reply:
Good idea, I will update it.